### PR TITLE
Upgrade org.xerial.snappy:snappy-java to 1.1.8.4

### PR DESCRIPTION
**Pull request by bradfordb-vmware**
Support for Apple Silicon was added in 1.1.8.2

Authored-by: Bradford D. Boyle <bradfordb@vmware.com>

### CODE REVIEW

**Review by ashuka24 — APPROVED** [review]
there was a PR for aws and now this one for snappy that provides support for Apple Silicon. Are there any other dependencies that we need to update that we could include in this PR?

File: server/build.gradle [hunk 1]
```diff
@@ -112,7 +112,7 @@ configure(javaProjects) {
             dependency("org.threeten:threeten-extra:1.5.0")
             dependency("org.tukaani:xz:1.8")
             dependency("org.wildfly.openssl:wildfly-openssl:1.0.7.Final")
-            dependency("org.xerial.snappy:snappy-java:1.0.5")
+            dependency("org.xerial.snappy:snappy-java:1.1.8.4")
```
Comment (denalex) [inline server/build.gradle:115]: a question to ask is : "If we use this newer library, compress the data with it and write the file to Hadoop, can a downstream consumer using default (presumably older) version included in Hadoop stack to properly decompress it ?"

Comment (bradfordb-vmware) [reply server/build.gradle:115]: I'm pretty sure the answer is yes. I used PXF (with this PR's change included) to write Parquet data with snappy compression to HDFS. I then used Hive 1.2.1000 (HDP, which includes snappy 1.0.5) to check if I could read the data and I was able to.

**Review by denalex — APPROVED** [review]
thank you for testing the change

**Pull request merged**